journey/pepr-deploy.ts

```diff
@@ -36,6 +38,32 @@ export function peprDeploy() {
 
   describe("should perform mutation of resources applied to the test cluster", testMutate);
 
+  describe("should monitor the cluster for admission changes", () => {
+    const command = 'npx';
+    const args = ['pepr', 'monitor', 'static-test'];
+    let childProcess: ChildProcess;
+
+    beforeAll(() => {
+      jest.setTimeout(5000);
+      childProcess = spawn(command, args, {
+        shell: true,
+      });
+    });
+
+    afterAll((done) => {
+      childProcess.kill("SIGKILL");
+      childProcess.removeAllListeners();
+      done()
+    });
+
+    it("npx pepr monitor should display results to console", async () => {
+      const output: string = await getMonitorData(childProcess)
+      const shouldPass = output.includes("✅") || output.includes("❌");
+      expect(shouldPass).toBe(true);
+    });
+  });
```

Comment:
```suggestion
  describe("should monitor the cluster for admission changes", () => {

    const until = (predicate: () => boolean ) : Promise<void> => {
      const poll = (resolve: () => void) => {
        if ( predicate() ) { resolve() }
        else { setTimeout(_ => poll(resolve), 250) }
      }
      return new Promise(poll);
    }

    it("npx pepr monitor should display validation results to console", async () => {
      const cmd = [ 'pepr', 'monitor', 'static-test' ]

      const proc = spawn('npx', cmd, { shell: true })

      const state = { accept: false, reject: false, done: false}
      proc.stdout.on('data', (data) => {
        const stdout: String = data.toString()
        state.accept = stdout.includes("✅") ? true : state.accept
        state.reject = stdout.includes("❌") ? true : state.reject

        if (state.accept && state.reject) {
          proc.kill()
          proc.stdin.destroy()
          proc.stdout.destroy()
          proc.stderr.destroy()
        }
      })
      proc.on('exit', () => state.done = true );

      await until(() => state.done)

      // completes only if conditions are met, so... getting here means success!
    }, 5000);
  });

```

Reply:
Something like this 1) kills cleanly, 2) doesn't have leave a hanging stdio pipe for Jest to cleanup, and 3) reads a bit easier (to me, at least) because it avoids the jest before/after lifecycle hooks for a single wait-until-seen test case.

Lot's of judgement-call type stuff in there, so happy to talk through any of it.